src/matrix/matrixnormal.jl

```diff
@@ -0,0 +1,159 @@
+"""
+    MatrixNormal{T <: Real, ST <: AbstractPDMat} <: ContinuousMatrixDistribution
+    
+Matrix Normal Distribution
+
+`X ~ MN(M, U, V)`
+
+M: n x p
+
+U: n x n positive definite
+
+V: p x p positive definite
+
+f(X) = c0 * exp( -0.5 tr[inv(V) (X - M)' inv(U) (X - M)] )
+
+c0   = (2pi) ^ {-np / 2} |V| ^ {-n / 2} |U| ^ {-p / 2}
+
+https://en.wikipedia.org/wiki/Matrix_normal_distribution
+
+"""
+struct MatrixNormal{T <: Real, ST <: AbstractPDMat} <: ContinuousMatrixDistribution
+
+    M::Matrix{T}
```

Comment:
Is there a reason we can't have `M` being a subtype of `AbstractMatrix`, rather than `Matrix`?

Reply:
agreed, we are trying to make Distributions as generic as possible when there are no hard constraints

Reply:
Changed.